### PR TITLE
Expose metrics for OPA authorizer

### PR DESCRIPTION
**Pull request by quangminhtran94**
Hi,
This PR is to expose some metrics for OPA authorizer, which currently doesn't expose any metric.
Please refer to this issue https://github.com/anderseknert/opa-kafka-plugin/issues/37

### CODE REVIEW

**Comment by anderseknert** [comment]
This is great! It’s in the middle of the night here but I’ll review this more carefully later today or tomorrow. The only issue I saw when looking briefly at this was that if we introduce new config options, we should mention these in the README. But this is **awesome**! Thank you so much 👍

**Comment by quangminhtran94** [comment]
Thank you @anderseknert , I have added documentation for metrics

**Review by scholzj — COMMENTED** [review]
I left one nit. I also wonder what would the performance impact on metrics be if enabled.

File: src/main/scala/org/openpolicyagent/kafka/OpaAuthorizer.scala [hunk 1]
```diff
@@ -32,10 +34,14 @@ class OpaAuthorizer extends Authorizer with LazyLogging {
   private lazy val allowOnError = config.getOrElse("opa.authorizer.allow.on.error", "false").toBoolean
   private lazy val superUsers = config.getOrElse("super.users", "").split(";").toList
 
+  private var metrics: Option[Metrics] = None
+
+  private lazy val maxCacheCapacity = config.getOrElse("opa.authorizer.cache.maximum.size", "50000").toInt
```
[the remark this answers is not in the document]
Comment (quangminhtran94) [reply src/main/scala/org/openpolicyagent/kafka/OpaAuthorizer.scala]: I have updated accrordingly

**Comment by quangminhtran94** [comment]
> I left one nit. I also wonder what would the performance impact on metrics be if enabled.

Thanks, I also had the same concern. I have run `OpaAuthorizerBenchmark`, the result between enable and disable metrics are nearly identical

**Review by anderseknert — APPROVED** [review]
LGTM! @scholzj if the changes look good to you we'll have this merged and I'll build a new release later today.

**Review by scholzj — APPROVED** [review]
Thanks for the PR.

**Pull request merged**